vms/platformvm/state/state.go

```diff
@@ -123,6 +124,8 @@ type Chain interface {
 
 	GetTx(txID ids.ID) (*txs.Tx, status.Status, error)
 	AddTx(tx *txs.Tx, status status.Status)
+
+	MerkleView() (merkledb.TrieView, error)
```

Comment:
It's quite annoying to expose this method since IIUC, diffs are basically immutable one we are done applying content from the block they referece.
So it'd be nice if we had a way to index these merkle views by blockID as we do with diffs in `state.Versions` and provide them in the diff constructor